Add pagination to project list

manthanank opened the pull request:
This pull request introduces pagination functionality to the projects panel on the website. It includes changes to both the HTML and JavaScript files to implement and handle pagination logic.

### Pagination functionality:

* [`index.html`](diffhunk://#diff-0eb547304658805aad788d320f10bf1f292797b5e6d745a3bf617584da017051R13-R17): Added pagination controls with "Previous" and "Next" buttons and a page information display.
* [`javascripts/app.js`](diffhunk://#diff-8b62c7fc620b4c7ea43b749165b75e583fbad4b4beb1c50c35bd7ddd1036f1bbL1-L6): Removed `@ts-nocheck` directive and added a function to handle pagination logic.

### JavaScript enhancements:

* [`javascripts/app.js`](diffhunk://#diff-8b62c7fc620b4c7ea43b749165b75e583fbad4b4beb1c50c35bd7ddd1036f1bbL18-R175): Updated the `renderProjects` function to render the correct page of projects and added event listeners for pagination controls.
* [`javascripts/app.js`](diffhunk://#diff-8b62c7fc620b4c7ea43b749165b75e583fbad4b4beb1c50c35bd7ddd1036f1bbL18-R175): Added functions to validate pagination logic and ensure buttons functionality.This pull request introduces pagination functionality to the projects panel in the web application. The key changes include adding pagination controls to the HTML, updating the JavaScript to handle pagination logic, and modifying the `renderProjects` function to support paginated data.

HTML changes:

* Added pagination controls (`Previous` and `Next` buttons) to the projects panel in `index.html`.

JavaScript changes:

* Removed `@ts-nocheck` directive from `javascripts/app.js`.
* Added a `paginateProjects` function to handle slicing the projects array based on the current page and limit.
* Updated the `renderProjects` function to render the correct page of projects and added logic to update pagination controls.
* Added event listeners for the pagination controls to handle page navigation.Related to #4918

Add pagination to the project list to improve navigation and reduce scroll time.

* Add pagination controls (Next, Previous) to `index.html` below the project list.
* Add a container for pagination controls in `index.html`.
* Add a function to handle pagination logic in `javascripts/app.js`.
* Update the `renderProjects` function in `javascripts/app.js` to render the correct page of projects.
* Add event listeners for pagination controls in `javascripts/app.js`.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/up-for-grabs/up-for-grabs.net/pull/4990?shareId=3a810bad-3ac2-4a9e-90b1-377465387464).

manthanank committed:
Add pagination to project list

Related to #4918

Add pagination to the project list to improve navigation and reduce scroll time.

* Add pagination controls (Next, Previous) to `index.html` below the project list.
* Add a container for pagination controls in `index.html`.
* Add a function to handle pagination logic in `javascripts/app.js`.
* Update the `renderProjects` function in `javascripts/app.js` to render the correct page of projects.
* Add event listeners for pagination controls in `javascripts/app.js`.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/up-for-grabs/up-for-grabs.net/issues/4918?shareId=XXXX-XXXX-XXXX-XXXX).

shiftkey commented:
Does this work as expected? I'm using [this link](https://deploy-preview-4990--up-for-grabs-test-bench.netlify.app/#/filters?names=&labels=&tags=javascript) to test pagination of `javascript` projects but it only renders as one page - and the buttons at the end of the form are disabled:

<img width="2096" alt="Screenshot 2024-12-30 at 3 15 21 PM" src="https://github.com/user-attachments/assets/108010a7-e5d4-44c5-8eab-c03925527881" />

manthanank committed:
Add pagination controls and logic to project list

* **index.html**
  - Add pagination controls (Next, Previous) below the project list
  - Add a container for pagination controls

* **javascripts/app.js**
  - Add a function to handle pagination logic
  - Update the renderProjects function to render the correct page of projects
  - Add event listeners for pagination controls

manthanank committed:
Add pagination controls and logic to project list

* **index.html**
  - Add pagination controls (Next, Previous) below the project list
  - Add a container for pagination controls

* **javascripts/app.js**
  - Add a function to handle pagination logic
  - Update the `renderProjects` function to render the correct page of projects
  - Add event listeners for pagination controls

manthanank committed:
Add pagination controls and logic to project list

* **index.html**
  - Add pagination controls (Next, Previous) below the project list
  - Add a container for pagination controls

* **javascripts/app.js**
  - Add a function to handle pagination logic
  - Update the `renderProjects` function to render the correct page of projects
  - Add event listeners for pagination controls
  - Fix the issue where next and previous buttons are disabled

manthanank committed:
Remove pagination logic from `javascripts/app.js`

* Remove the function to handle pagination logic
* Update the `renderProjects` function to remove pagination logic
* Remove event listeners for pagination controls

manthanank committed:
Add pagination controls and logic to project list

* **index.html**
  - Add pagination controls (Next, Previous) below the project list
  - Add a container for pagination controls

* **javascripts/app.js**
  - Add a function to handle pagination logic
  - Update the `renderProjects` function to include pagination logic
  - Add event listeners for pagination controls

manthanank committed:
Add pagination controls and logic to project list

* **index.html**
  - Add pagination controls (Next, Previous) below the project list
  - Add a container for pagination controls
* **javascripts/app.js**
  - Add a function to handle pagination logic
  - Update the `renderProjects` function to render the correct page of projects
  - Add event listeners for pagination controls

manthanank committed:
Add pagination controls and logic to project list

* **index.html**
  - Add pagination controls (Next, Previous) below the project list
  - Add a container for pagination controls

* **javascripts/app.js**
  - Add a function to handle pagination logic
  - Update the `renderProjects` function to render the correct page of projects
  - Add event listeners for pagination controls
  - Fix the issue of showing more than 15 cards per page
  - Add a function to validate pagination logic
  - Add a function to ensure buttons functionality works properly
  - Add logic to disable/undisable buttons based on data
